zamba/models/config.py

```diff
@@ -0,0 +1,81 @@
+from enum import Enum
+from pathlib import Path
+from typing import Optional, Union
+
+from pydantic import BaseModel as PydanticBaseModel
+from pydantic import DirectoryPath, FilePath
+import yaml
+
+
+class BaseModel(PydanticBaseModel):
+    """Set defaults for all models that inherit from the pydantic base model."""
+
+    class Config:
+        extra = "forbid"
+        use_enum_values = True
+
+
+class ModelClassEnum(str, Enum):
+    cnnensemble = "cnnensemble"
+    custom = "custom"
+    sample = "sample"
+
+
+class ModelLibraryEnum(str, Enum):
+    keras = "keras"
+    pytorch = "pytorch"
+
+
+class ModelProfileEnum(str, Enum):
+    fast = "fast"
+    full = "full"
+
+
+class RegionEnum(str, Enum):
+    us = "us"
+    eu = "eu"
+    asia = "asia"
+
+
+class TrainConfig(BaseModel):
+    train_data: DirectoryPath = None
+    val_data: DirectoryPath = None
+    labels: FilePath = None
+    model_path: FilePath = None
+    model_library: ModelLibraryEnum = "keras"
+    model_class: ModelClassEnum = "custom"
+    tempdir: Optional[Path] = None
+    n_epochs: Optional[int] = 10
+    height: Optional[int] = None
+    width: Optional[int] = None
+    augmentation: Optional[bool] = False
+    early_stopping: Optional[bool] = False
```

Comment:
Let's pull out the unused features for now like the two above and document them in an issue. Better to track that potential work than to remove these later (which would break configs that specify these items)

Reply:
dropped into https://github.com/drivendataorg/zamba-algorithms/issues/25